### ISSUE TITLE
Image caption not send

### ISSUE THREAD

**Issue by a7lan** [tgalal/yowsup#1628]
Image send without caption. I used comand

/image send XXXXXXXXXXX /tmp/image.png caption=test
and
/image send XXXXXXXXXXX /tmp/image.png test

Image received, but withou caption

**Comment by marcoarildo** [comment]
a7lan, hi

Did you get send image without caption ???

Hugs

**Comment by a7lan** [comment]
Yes. But I do not use anymore. Whatsapp three times blocked me. Now I use Telegram

**Comment by marcoarildo** [comment]
@a7lan Take a look in the file /yowsup/demos/cli/layer.py. The parameters "caption" is **NONE**.
You have the fix this.. **remove** the "None"

def image_send(self, number, path, caption = None):
...
def media_send(self, number, path, mediaType, caption = None):
...
successFn = lambda successEntity, originalEntity: self.onRequestUploadResult(jid, mediaType, 
path, successEntity, originalEntity, caption)

Good luck

**Comment by Dargmuesli** [comment]
I think if you call `image_send` with `caption="xyz"` you do not have to remove the `None` there.
But nevertheless at least videos do not support captions currently.

**Comment by gushome7** [comment]
Please note on cli.py:
    @clicmd("Send an image with optional caption")
    def image_send(self, number, path, caption = None):
This line ==>  self.media_send(number, path, RequestUploadIqProtocolEntity.MEDIA_TYPE_IMAGE)
Don't include caption param. Need to add it:

    @clicmd("Send an image with optional caption")
    def image_send(self, number, path, caption = None):
        self.media_send(number, path, RequestUploadIqProtocolEntity.MEDIA_TYPE_IMAGE,caption)
